src/index.js

```diff
@@ -94,6 +102,18 @@ export default (inputs) => {
     }
 
     try {
+      if (schema) {
+        await handleErrorWithSource(
+          "schema",
+          schema.validateAsync({
```

Comment:
I don't really understand how the schema is working, roles. e.g.
Add an example with an explanation in the README.